dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -208,6 +237,7 @@ public BinaryRecordReader(InputStream is, long streamPos) throws Exception
 
         this.inputStream = new CountingInputStream(is);
         this.inputStream.streamPos = streamPos;
+        this.recordBuilderSpan = recBuildSpan;
```

Comment:
I think recordBuilderSpan is acting as a parent span for all potential recordBatchSpans, is it tracking anything else? and why is it passed in, what is the caller tracking in this span